cv-application/src/App.jsx

```diff
@@ -1,35 +1,83 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import React, { useState } from "react";
+import GeneralInfo from "./components/GeneralInfo";
+import Education from "./components/Education";
+import Experience from "./components/Experience";
+import CVPreview from "./components/CVPreview";
+import "./styles/app.css";
+import "./styles/preview.css";
 
-function App() {
-  const [count, setCount] = useState(0)
+export default function App() {
+  const [general, setGeneral] = useState({ name: "", email: "", phone: "" });
+  const [educationList, setEducationList] = useState([]);
+  const [experienceList, setExperienceList] = useState([]);
+
+  const handleGeneralSubmit = (data) => setGeneral(data);
+
+  // Education handlers
+  const addEducation = (edu) => {
+    setEducationList((prev) => [
+      ...prev,
+      { ...edu, id: Date.now().toString() },
+    ]);
+  };
+
+  const updateEducation = (id, edu) => {
+    setEducationList((prev) =>
+      prev.map((e) => (e.id === id ? { ...e, ...edu } : e))
+    );
+  };
+
+  const removeEducation = (id) => {
+    setEducationList((prev) => prev.filter((e) => e.id !== id));
+  };
+
+  // Experience handlers
+  const addExperience = (exp) => {
+    setExperienceList((prev) => [
+      ...prev,
+      { ...exp, id: Date.now().toString() },
```

Comment:
Don't you think this method { ...exp, id: crypto.randomUUID() } will handle the scenario where two items are added quickly